packages/std/src/common/denom.rs

```diff
@@ -1,7 +1,41 @@
-use crate::error::ContractError;
-use cosmwasm_std::{ensure, Deps};
+use crate::{ado_contract::ADOContract, amp::AndrAddr, error::ContractError};
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, Deps, DepsMut, Env};
 pub const SEND_CW20_ACTION: &str = "SEND_CW20";
 
+#[cw_serde]
+pub enum Asset {
+    Cw20Token(AndrAddr),
+    NativeToken(String),
+}
+impl Asset {
+    pub fn get_verified_asset(
+        &self,
+        deps: DepsMut,
+        env: Env,
+    ) -> Result<(String, bool), ContractError> {
+        match self {
+            Asset::Cw20Token(cw20_token) => {
+                let cw20_token = cw20_token.get_raw_address(&deps.as_ref())?;
```

Comment:
We probably want a way of verifying if the provided address is a cw20 contract (to the best of our ability). Maybe querying [this](https://github.com/andromedaprotocol/andromeda-core/blob/main/packages/andromeda-fungible-tokens/src/cw20.rs#L190) might be better than nothing.

Reply:
Should we ensure that the supply is above zero or be satisfied with the query not erroring?